### ISSUE TITLE
xarray spatial look for cuda with Dask xarray

### ISSUE THREAD

**Issue by srggrs** [makepath/xarray-spatial#571]
**Describe the bug**
I have a dask xarray that I read using the `chunks` argument using the [`rioxarray`](https://corteva.github.io/rioxarray/stable/) library. When I call the `focal_stats` or `apply` function from `xrspatial.focal` I get a CUDA initialisation e related error.

**Expected behavior**
I expect the computation will run using the Dask backend on CPU and not CUDA

**Screenshots**
Stack trace after calling `focal_stats`:

```bash
2021-11-08 17:33:58,068 driver INFO : init
2021-11-08 17:34:06,050 driver ERROR : Call to cuInit results in CUDA_ERROR_NO_DEVICE
```

**Desktop (please complete the following information):**
 - OS: Ubuntu 18.04 LTS
- rioxarray                 0.8.0
- xarray                    0.19.0
- xarray-spatial            0.2.9

**Additional context**
My laptop has a NVIDA with CUDA enabled but I need to run on Dask backend so that I can parallelise on HPC.

__UPDATE 2021-11-09:__ my colleague running the same code on his MAC (with no NVIDIA GPU) does not have this issue, and raster Dask array is computed on CPU. There is a way to disable detection of the GPU? I can link code (is not long at all) here (e.g. via GitHub gist)

__UPDATE 2021-11-11:__ I can confirm my code is not the issue, as I managed to run the very same code within a Docker container on my machine with Dask backend without incurring in this error

**Comment by ianthomas23** [comment]
@srggrs Thanks for reporting this. It is part of a wider problem that we need to consistent identify and handle the presence or absence of cupy and cuda as well as the user's preferences. I expect progress on this in the next few weeks.

**Comment by thuydotm** [comment]
Fixed since #612
